.ado/scripts/prepublish-check.mjs

```diff
@@ -0,0 +1,195 @@
+// @ts-check
+import { spawnSync } from "node:child_process";
+import * as fs from "node:fs";
+import * as util from "node:util";
+
+/**
+ * @typedef {typeof import("../../nx.json")} NxConfig
+ * @typedef {{ tag?: string }} Options
+ */
+
+/**
+ * Exports a variable, `publish_react_native_macos`, to signal that we want to
+ * enable publishing on Azure Pipelines.
+ *
+ * Note that pipelines need to read this variable separately and do the actual
+ * work to publish bits.
+ */
+function enablePublishingOnAzurePipelines() {
+  console.log(`##vso[task.setvariable variable=publish_react_native_macos]1`);
+}
+
+/**
+ * Returns whether the given branch is considered main branch.
+ * @param {string} branch
+ */
+function isMainBranch(branch) {
+  // There is currently no good way to consistently get the main branch. We
+  // hardcode the value for now.
+  return branch === "main";
+}
+
+/**
+ * Returns whether the given branch is considered a stable branch.
+ * @param {string} branch
+ */
+function isStableBranch(branch) {
+  return /^\d+\.\d+-stable$/.test(branch);
+}
+
+/**
+ * Loads Nx configuration.
+ * @returns {NxConfig}
+ */
+function loadNxConfig(configFile = "nx.json") {
+  const nx = fs.readFileSync(configFile, { encoding: "utf-8" });
+  return JSON.parse(nx);
+}
+
+/**
+ * Returns a numerical value for a given version string.
+ * @param {string} version
+ * @returns {number}
+ */
+function versionToNumber(version) {
+  const [major, minor] = version.split("-")[0].split(".");
+  return Number(major) * 1000 + Number(minor);
+}
+
+/**
+ * Returns the currently checked out branch. Note that this function prefers
+ * predefined CI environment variables over local clone.
+ * @returns {string}
+ */
+function getCurrentBranch() {
+  // https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services
+  const adoSourceBranchName = process.env["BUILD_SOURCEBRANCHNAME"];
+  if (adoSourceBranchName) {
+    return adoSourceBranchName.replace(/^refs\/heads\//, "");
+  }
+
+  // Depending on how the repo was cloned, HEAD may not exist. We only use this
+  // method as fallback.
+  const { stdout } = spawnSync("git", ["rev-parse", "--abbrev-ref", "HEAD"]);
+  return stdout.toString().trim();
+}
+
+/**
+ * Returns the latest published version of `react-native-macos` from npm.
+ * @returns {number}
+ */
+function getLatestVersion() {
+  const { stdout } = spawnSync("npm", ["view", "react-native-macos@latest", "version"]);
```

Comment:
Man I've been looking for nice NPM libraries to interact with the registry, when instead I could have been using spawn 😂